frontend/src/scenes/plugins/Plugins.tsx

```diff
@@ -47,14 +48,16 @@ export function Plugins(): JSX.Element {
 
             {user.team?.plugins_opt_in ? (
                 <>
-                    {user.plugin_access.install ? (
+                    {user.organization?.plugins_access_level >= PluginsAccessLevel.Install ? (
                         <Tabs activeKey={pluginTab} onChange={(activeKey) => setPluginTab(activeKey as PluginTab)}>
                             <TabPane tab="Installed" key={PluginTab.Installed}>
                                 <InstalledTab />
                             </TabPane>
-                            <TabPane tab="Repository" key={PluginTab.Repository}>
-                                <RepositoryTab />
-                            </TabPane>
+                            {user.organization?.plugins_access_level === PluginsAccessLevel.Root && (
```

Comment:
This code is not self-evident, it's not clear to a reader what `plugin_access_level` stands for or what the right levels are.

Suggestion:

Extract module `frontend/src/scenes/plugins/accessControl.ts` containing function(s) like `hasAccessToPlugins(user)`,  `canInstallPlugins(user)` etc. This keeps the business logic centralized and the component files readable.

Reply:
Done